File: Q32/Q32.java
import java.util.Scanner;

public class Q32 {

	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int num;
		// random symbols for pattern
		String[] symbols = {"$", "*", "@", "#", "%"};
		
		while (true) {
			System.out.println("Enter any number: ");
			num = input.nextInt();
			if (num > 0)
				break;
			System.out.println("Invalid input, try again:");
		}


        // outer loop is for the number of rows
        // the 1st and 3rd inner loop are for the number of spaces per row
        // the 2nd inner loop is for the number or random symbols
        for (int i = num; i > 0; i--) {
        	int random = (int) (Math.random()*symbols.length);
            for (int j = i - 1; j > 0; j--)
                System.out.print(" ");
            for (int k = (num - i + 1) * 2 - 1; k > 0; k--)
                System.out.print(symbols[random]);
            for (int j = i - 1; j > 0; j--)
                System.out.print(" ");
            System.out.println();
        }
		
		


}

}

File: Q32/Q32.py
import random

symbols = ["$", "*", "@", "#", "%"]

def printPattern(rows):
    for rowIndex in range(rows):
        randomIndex = random.randint(0, len(symbols) - 1)
        for spaces in range(rows - rowIndex - 1):
            print(" ", end="")
        for symbol in range(2*rowIndex + 1):
            print(symbols[randomIndex], end="")
        for spaces in range(rows - rowIndex - 1):
            print(" ", end="")
        print()


if __name__ == '__main__':
    userInput = int(input("Number of Rows: "))
    printPattern(userInput)